### ISSUE TITLE
[Bug]: chrome.tabs.update rejected with "Navigation rejected." on Chrome 152 when the extension detaches the debugger right before navigating

### ISSUE THREAD

**Issue by dirsen-gan** [jackwener/opencli#2487]
### What happened

  Any OpenCLI adapter that only navigates (no network capture) fails on Chrome 152 with:

  ```
  ok: false
  error:
    code: UNKNOWN
    message: Navigation rejected.
    exitCode: 1
  ```

  `Navigation rejected.` is Chrome's own error — the string is present in the Chrome 152 binary — i.e.
  `chrome.tabs.update(tabId, { url })` is being rejected.

  ### Environment

  - macOS (Darwin 25.2.0), arm64
  - Google Chrome 152.0.7977.83
  - OpenCLI CLI 1.8.6 (OpenCLIApp 0.1.37)
  - Browser Bridge extension 1.0.24 (ildkmabpimmkaediidaifkhjpohdnifk)

  ### Root cause

  In the extension bundle (`dist/background.js`, v1.0.24), `handleNavigate` does:

  ```js
  if (!hasActiveNetworkCapture(tabId)) {
      await detach(tabId);                 // chrome.debugger.detach({ tabId })
  }
  await chrome.tabs.update(tabId, { url: targetUrl });   // rejected by Chrome 152
  ```

  With `--trace` (`on` / `retain-on-failure`), `execution.js` calls `page.startNetworkCapture()`, so
  `hasActiveNetworkCapture(tabId)` is true and the `detach()` is skipped → navigation succeeds. With `--trace` off, the
  detach runs and the immediately following `chrome.tabs.update` is rejected.

  ### Expected

  `navigate` succeeds regardless of `--trace`.

  ### Actual

  `chrome.tabs.update` rejects navigation with `Navigation rejected.` whenever the debugger was detached immediately
  before. The failure is persistent per profile: once a profile's automation tab is left in the "previously attached"
  state, every plain invocation keeps failing.

  ### Workaround

  Keep a network capture active before navigating (what trace mode does internally):

  ```js
  await page.startNetworkCapture?.();
  await page.goto(probeUrl);

### Steps to Reproduce

1. Use any adapter that calls `page.goto(...)` and does NOT enable network capture.
  2. Run it without trace mode — it fails:
     ```
     opencli --profile <profile> <site> <command> ... --format json
     ```
     → exit 1, `message: Navigation rejected.`
  3. Run the SAME command with trace mode — it succeeds:
     ```
     opencli --profile <profile> <site> <command> ... --format json --trace retain-on-failure
     ```
     → exit 0, command completes.
  4. Run it again without trace mode → it fails again.

  A/B on the same profile, back to back:

  | run | flags | result |
  |-----|-------|--------|
  | 1 | (none) | FAIL |
  | 2 | (none) | FAIL |
  | 3 | `--trace retain-on-failure` | SUCCESS |
  | 4 | (none) | FAIL |

  The only variable is whether the debugger stays attached (trace mode calls `page.startNetworkCapture()`), which makes
  the extension skip its pre-navigation `detach()`.

[repro.sh](https://github.com/user-attachments/files/32085162/repro.sh)

### Expected Behavior

`navigate` / `page.goto(...)` should succeed regardless of whether `--trace` is enabled. `chrome.tabs.update` should
  not be rejected merely because the debugger was detached immediately before it.  

### OpenCLI Version

1.8.6    

### Node.js Version

22.x

### Operating System

macOS

### Logs / Screenshots

```shell

```

**Comment by Wood9264** [comment]
### Additional data point: also reproducible on Microsoft Edge 153 (Chromium), plus a proposal to sink the fix into shared layers

Thanks for the detailed diagnosis — the `detach()` → `chrome.tabs.update()` ordering reproduces here as well, and it is not Chrome-specific:

| Item | Value |
|---|---|
| Browser | Microsoft Edge 153.0.4234.32 (`Navigation rejected.` is present in `msedge.dll`) |
| CLI / extension | opencli 1.8.6 / Browser Bridge 1.0.24 |
| Affected code path | `google search`, `duckduckgo search`, `yahoo search`, `brave search` — all import `runBrowserStep` from `clis/_shared/search-adapter.js` and call `page.goto()` directly |
| Not affected | `opencli browser <session> open <same-url>` (browser-surface session, the adapter window/tab is never left in the "previously attached" state) |
| A/B | plain run fails → `--trace retain-on-failure` succeeds → plain run fails again. On Edge the failure is intermittent per tab state rather than 100% per profile, but the pattern matches your table exactly |

Also worth noting: `clis/google/images.js` already carries a per-adapter workaround (`isNavigationRejected` → close window → retry → fall back to `newTab`), and the facebook `marketplace-search` PR mentions the same pattern. So the workaround exists, it just was never sunk to a shared layer.

### Proposal: sink the fix, don't duplicate it per adapter

#### 1. Bridge layer (preferred primary fix) — `src/browser/page.ts`, `Page.goto()`

`goto()` already retries once for the stale-page-identity case. Add a second, narrower retry for `Navigation rejected.`:

```ts
function isNavigationRejectedError(err: unknown): boolean {
    return /Navigation rejected/i.test(err instanceof Error ? err.message : String(err));
}

// inside Page.goto()
try {
    result = await sendCommandFull('navigate', { url, ...this._cmdOpts() });
} catch (err) {
    if (isNavigationRejectedError(err)) {
        // Chrome 152+: a navigate issued right after chrome.debugger.detach() can be
        // rejected at the content layer. "Navigation rejected." means no navigation
        // handle was ever created, so a retry is safe (this is not an unknown-outcome write).
        await new Promise((resolve) => setTimeout(resolve, 250));
        result = await sendCommandFull('navigate', { url, ...this._cmdOpts() });
    } else if (isStalePageIdentityError(err) && this._page !== undefined) {
        this._page = undefined;
        result = await sendCommandFull('navigate', { url, ...this._cmdOpts() });
    } else {
        throw err;
    }
}
```

Why here first:

- covers every adapter **and** every `opencli browser` command in one place;
- ships with the npm package, so users get it without waiting for an extension store release;
- the reporter's workaround (`startNetworkCapture` before `goto`) shows the retry lands after the detach has settled, so the second attempt succeeds;
- retry is semantically safe: the error is thrown synchronously by `tabs.update` before a navigation handle exists, so nothing has been applied yet.

#### 2. Extension layer (root cause) — `handleNavigate`

The real fix is to stop racing `detach()` against `tabs.update()`, e.g. retry the update once on rejection:

```js
if (!hasActiveNetworkCapture(tabId)) {
    await detach(tabId);
}
await updateTabWithRetry(tabId, targetUrl);

async function updateTabWithRetry(tabId, url) {
    for (let attempt = 0; attempt < 2; attempt += 1) {
        try {
            await chrome.tabs.update(tabId, { url });
            return;
        } catch (err) {
            if (attempt === 0 && /Navigation rejected/i.test(String(err?.message ?? err))) {
                await new Promise((resolve) => setTimeout(resolve, 250));
                continue;
            }
            throw err;
        }
    }
}
```

This addresses the root cause for all commands, but needs an extension release, so it should not be the only fix.

#### 3. Adapter shared layer (stopgap + de-duplication) — `clis/_shared/search-adapter.js`

Move the `google/images.js` logic into a shared helper and switch the search adapters to it:

```js
// clis/_shared/search-adapter.js
export function isNavigationRejected(error) {
    return /Navigation rejected/i.test(String(error?.message || error));
}

export async function navigateWithRetry(page, url) {
    try {
        await page.goto(url);
        return;
    } catch (error) {
        if (!isNavigationRejected(error)) throw error;
    }
    if (typeof page.closeWindow === 'function') {
        await page.closeWindow().catch(() => {});
    }
    try {
        await page.goto(url);
        return;
    } catch (error) {
        if (!isNavigationRejected(error)) throw error;
        if (typeof page.newTab === 'function' && typeof page.setActivePage === 'function') {
            const pageId = await page.newTab(url);
            if (pageId) {
                await page.setActivePage(pageId);
                return;
            }
        }
        throw error;
    }
}
```

Then in `clis/google/search.js`, `clis/duckduckgo/search.js`, `clis/brave/search.js`, `clis/yahoo/search.js`:

```js
await runBrowserStep('brave search navigation', () => navigateWithRetry(page, url));
```

Once the helper exists, `clis/google/images.js` (and the facebook PR) should import it instead of keeping local copies — otherwise we end up with three copies of the same workaround.
